src/Features/Lsif/Generator/Program.cs

```diff
@@ -27,20 +25,41 @@ internal static class Program
+            var output = new CliOption<string?>("--output") { Description = "file to write the LSIF output to, instead of the console", DefaultValueFactory = _ => null };
+            output.AcceptLegalFilePathsOnly();
+            var outputFormat = new CliOption<LsifFormat>("--output-format") { Description = "format of LSIF output", DefaultValueFactory = _ => LsifFormat.Line };
+            var log = new CliOption<string?>("--log") { Description = "file to write a log to", DefaultValueFactory = _ => null };
+            log.AcceptLegalFilePathsOnly();
+
+            var generateCommand = new CliRootCommand("generates an LSIF file")
             {
-                new Option("--solution", "input solution file") { Argument = new Argument<FileInfo>().ExistingOnly() },
-                new Option("--project", "input project file") { Argument = new Argument<FileInfo>().ExistingOnly() },
-                new Option("--compiler-invocation", "path to a .json file that contains the information for a csc/vbc invocation") { Argument = new Argument<FileInfo>().ExistingOnly() },
-                new Option("--binlog", "path to a MSBuild binlog that csc/vbc invocations will be extracted from") { Argument = new Argument<FileInfo>().ExistingOnly() },
-                new Option("--output", "file to write the LSIF output to, instead of the console") { Argument = new Argument<string?>(defaultValue: () => null).LegalFilePathsOnly() },
-                new Option("--output-format", "format of LSIF output") { Argument = new Argument<LsifFormat>(defaultValue: () => LsifFormat.Line) },
-                new Option("--log", "file to write a log to") { Argument = new Argument<string?>(defaultValue: () => null).LegalFilePathsOnly() }
+                solution,
+                project,
+                compilerInvocation,
+                binLog,
+                output,
+                outputFormat,
+                log
             };
 
-            generateCommand.Handler = CommandHandler.Create((Func<FileInfo?, FileInfo?, FileInfo?, FileInfo?, string?, LsifFormat, string?, Task>)GenerateAsync);
-
-            return generateCommand.InvokeAsync(args);
+            generateCommand.SetAction((parseResult, cancellationToken) =>
+            {
+                return GenerateAsync(
+                    solution: parseResult.GetValue(solution),
+                    project: parseResult.GetValue(project),
+                    compilerInvocation: parseResult.GetValue(compilerInvocation),
+                    binLog: parseResult.GetValue(binLog),
+                    output: parseResult.GetValue(output),
+                    outputFormat: parseResult.GetValue(outputFormat),
+                    log: parseResult.GetValue(log),
+                    cancellationToken);
+            });
+
+            return generateCommand.Parse(args).InvokeAsync(CancellationToken.None);
```

Comment:
Same question as earlier: what should go here for the token?

Reply:
> Same question as earlier: what should go here for the token?

If this is `Main`, then `None`.

If this was any other async method that would have it's own token, then that token.